### PR TITLE
--[Draft/WIP] Scene Dataset and Scene Import docs and tutorials

### PR DESCRIPTION
**Pull request by jturner65**
## Motivation and Context
This DRAFT PR introduces the documentation for Scene Datasets and Scene Import, as well as python/colab tutorial on Scene Dataset and Scene Instance configuration and usage.

## Types of changes

- [ ] Docs change / refactoring / dependency upgrade
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist

- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have completed my CLA (see **CONTRIBUTING**)
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: docs/pages/attributesJSON.rst [hunk 1]
```diff
@@ -29,7 +29,7 @@ Below are the supported JSON tags for Physics Manager Attributes templates, and
 
 "physics_simulator"
 	- string
-	- What physics engine should be used for dynamics simulation.  Currently supports "bullet" for Bullet physics simulation, and "none", meaning kinematic motion is to be used.
+	- What physics engine should be used for dynamics simulation. Currently supports "bullet" for Bullet physics simulation, and "none", meaning kinematic motion is to be used.
```
Comment (dhruvbatra) [inline docs/pages/attributesJSON.rst:32]: Can we add a sentence here saying what the implications of using kinematic motion are? Collisions still happen, but you can't apply forces.

File: docs/pages/attributesJSON.rst [hunk 2]
```diff
@@ -29,7 +29,7 @@ Below are the supported JSON tags for Physics Manager Attributes templates, and
...
 "gravity"
 	- 3-vector
 	- The default gravity to use for physical modeling. This can be overridden by Stage attributes.
```
Comment (dhruvbatra) [inline docs/pages/attributesJSON.rst:35]: And where do we describe what a "Stage" is?

File: docs/pages/attributesJSON.rst [hunk 3]
```diff
@@ -29,7 +29,7 @@ Below are the supported JSON tags for Physics Manager Attributes templates, and
 
 "physics_simulator"
 	- string
-	- What physics engine should be used for dynamics simulation.  Currently supports "bullet" for Bullet physics simulation, and "none", meaning kinematic motion is to be used.
+	- What physics engine should be used for dynamics simulation. Currently supports "bullet" for Bullet physics simulation, and "none", meaning kinematic motion is to be used.
 "gravity"
 	- 3-vector
```
Comment (dhruvbatra) [inline docs/pages/attributesJSON.rst:34]: Can we either say what the 3 dimensions correspond to or link to Bullet docs?

File: docs/pages/attributesJSON.rst [hunk 4]
```diff
@@ -46,9 +46,125 @@ Below are the supported JSON tags for Physics Manager Attributes templates, and
 	- list of strings
 	- A list of locations to query for supported object files that should be available to be loaded into the world.
```
Comment (dhruvbatra) [inline docs/pages/attributesJSON.rst:47]: Maybe point to another page referring to what kind of object files are supported.

File: docs/pages/attributesJSON.rst [hunk 5]
```diff
@@ -46,9 +46,125 @@ Below are the supported JSON tags for Physics Manager Attributes templates, and
...
 
+
+`Scene Dataset Attributes`_
+===========================
+Scene Dataset Attributes describe the names and file locations of all the assets and configuration files that make up a Scene Dataset. They are designed as a kind of index for a scene dataset, organizing the links to all the constituent components of the dataset into appropriate thematic categories, as well as providing an easy mechanism for customizing any of the configurations, all in a single file. All file paths referenced in this configuration should be relative to the directory where this Scene Dataset configuration resides.
```
Comment (dhruvbatra) [inline docs/pages/attributesJSON.rst:52]: And what is a scene dataset?

Comment (dhruvbatra) [reply docs/pages/attributesJSON.rst:52]: What we need is a glossary -- basically a list of terms we use and what they mean.

File: docs/pages/attributesJSON.rst [hunk 6]
```diff
@@ -46,9 +46,125 @@ Below are the supported JSON tags for Physics Manager Attributes templates, and
...
+
+`An example of an appropriately configured Scene Dataset Configuration file can be found below <../../../data/test_assets/dataset_tests/test_dataset.scene_dataset_config.json>`_:
+
+.. include:: ../../data/test_assets/dataset_tests/test_dataset.scene_dataset_config.json
+    :code: json
+
+'Scene Dataset: Stages, Objects, Light Setups and Scene Instances'
+------------------------------------------------------------------
+
+The Scene Dataset Attributes configuration consists of a hierarchy of JSON components used to specify all the component assets and attributes used within a scene dataset. Tags "stages", "objects", "light_setups" and "scene_instances" reference JSON objects that list the included assets and configurations for the specified type of construction. The JSON objects referenced by each of these 4 tags support the following tags:
```
Comment (dhruvbatra) [inline docs/pages/attributesJSON.rst:66]: Same thought as above -- we should define what a stage is and what a "scene instance" is.

File: docs/pages/attributesJSON.rst [hunk 7]
```diff
@@ -46,9 +46,125 @@ Below are the supported JSON tags for Physics Manager Attributes templates, and
...
+
+    "attributes"
+    - JSON object
+    - This tag references a configuration appropriate for the construction type. If "original_file" is specified, all the values from "original_file" are used, and overridden by any specified in this tag.  If "template_handle" is specified, the new attributes are saved with the specified "template_handle", otherwise they are saved with the "original_file" handle. See the individual attributes types' documentation for supported tags and format.
+
+'Scene Dataset: Navmeshes and Semantic Scene Descriptors'
+---------------------------------------------------------
+
+The Scene Dataset Attributes configuration also contains references to the navmeshes, using the JSON tag "navmesh_instances", and semantic scene descriptor files, using the tag "semantic_scene_descriptor_instances", that are available to the Scene Dataset.  These tags each reference a JSON object consisting of key-value pairs where the keys are the aliases to use in scene instances for a particular navmesh or semantic scene descriptor, and the values are the paths to the navmesh or SSD file, respectively.
```
Comment (dhruvbatra) [inline docs/pages/attributesJSON.rst:95]: We should hyperlink Navmeshes to the colab explaining Navmeshes. 

And what is a semantic scene descriptor file?

File: docs/pages/attributesJSON.rst [hunk 8]
```diff
@@ -46,9 +46,125 @@ Below are the supported JSON tags for Physics Manager Attributes templates, and
...
+
+.. include:: ../../data/test_assets/dataset_tests/scenes/dataset_test_scene.scene_instance.json
+    :code: json
+
+'Scene Instance Parameters'
+---------------------------
+
+"translation_origin"
+  - string
+  - Specifies scene instance default for translation origin. Values are case-insensitive. Supports "asset_local", where the origin is defined in the individual asset's local frame, "com", where the asset's com is the origin for transformations, or "unknown" if the value is unknown. This value can be overridden for the stage and any object instances in the scene.
```
Comment (dhruvbatra) [inline docs/pages/attributesJSON.rst:115]: Say that com stands for center of mass.

File: docs/pages/attributesJSON.rst [hunk 9]
```diff
@@ -46,9 +46,125 @@ Below are the supported JSON tags for Physics Manager Attributes templates, and
...
+
+"default_lighting"
+  - string
+  - The handle of the lighting setup configuration, defined in the Scene Dataset Configuration, that should be usedd when instantiating lights in the scene.
+
+"navmesh_instance"
+  - string
+  - The alias of the scene's navmesh asset, which is mapped to a file handle in the Scene Dataset Configuration.
```
Comment (dhruvbatra) [inline docs/pages/attributesJSON.rst:123]: And how does one generate a navmesh if one doesn't have it?

File: docs/pages/attributesJSON.rst [hunk 10]
```diff
@@ -46,9 +46,125 @@ Below are the supported JSON tags for Physics Manager Attributes templates, and
...
+
+The JSON tags the Scene Object Instance Attributes support are as follows:
+
+"template_name"
+  - string
+  - Dataset-unique name of the template for the stage or object attributes desired to instantiate. Since attributes templates can encode full file paths, their handles can be very long; therefore, the "template_name" field supports substring lookup, and an incomplete handle can be used for this field so long as it uniquely maps to the full handle specified in the Scene Dataset Attributes.
+
+"translation_origin"
+  - string
+  - Specifies the translation origin to use for the instantiation of the specified object. Values are case-insensitive. Supports "asset_local", where the origin is defined in the individual asset's local frame, "com", where the asset's com is the origin for transformations, or "unknown" if the value is unknown. If this value is specified, and not set to "unknown", it will override the top-level default, if any exists. Currently stages do not support instantiation transformations.
```
Comment (dhruvbatra) [inline docs/pages/attributesJSON.rst:150]: Same comment about COM as above.

File: docs/pages/attributesJSON.rst [hunk 11]
```diff
@@ -46,9 +46,125 @@ Below are the supported JSON tags for Physics Manager Attributes templates, and
...
+
+"motion_type"
+  - string
+  - The desired motion type of the object being instanced. Values are case-insensitive. Values supported are "static", "kinematic", and "dynamic". This field is ignored for stages, which are always created with STATIC motion type.
```
Comment (dhruvbatra) [inline docs/pages/attributesJSON.rst:154]: Link to somewhere where we describe what "static", "kinematic", and "dynamic" mean.

File: docs/pages/attributesJSON.rst [hunk 12]
```diff
@@ -46,9 +46,125 @@ Below are the supported JSON tags for Physics Manager Attributes templates, and
...
+
+"rotation"
+  - Quaternion (read as [w,x,y,z], with idx 0 referencing the scalar part.)
+  - The rotation to be applied to the object upon instantiation. Currently stages do not support instantiation transformations.
+
+
 `Stage Attributes`_
 ===================
-A stage in Habitat-Sim is a static object consisting of static background scenery wherein an agent acts.  Stage Attributes templates hold relevant information describing a stage's render and collision assets and physical properties.  Any source configuration files used to build these attributes should be named using the following format:
+A stage in Habitat-Sim is a static object consisting of static background scenery wherein an agent acts. Stage Attributes templates hold relevant information describing a stage's render and collision assets and physical properties. Any source configuration files used to build these attributes should be named using the following format:
```
Comment (dhruvbatra) [inline docs/pages/attributesJSON.rst:167]: Ah, there's the definition I was looking for. We should accumulate all such definitions and list them at one place.

File: docs/pages/attributesJSON.rst [hunk 13]
```diff
@@ -124,9 +239,10 @@ Below are stage-specific physical and object-related quantities.  These values w
 	- boolean
 	- Whether the stage should be rendered with lighting or flat shading.
 
+
 `Object Attributes`_
 ====================
-Object Attributes templates hold descriptive information for instancing rigid objects into Habitat-Sim.  These file names should be formatted as follows:
+Object Attributes templates hold descriptive information for instancing rigid objects into Habitat-Sim. These file names should be formatted as follows:
```
Comment (dhruvbatra) [inline docs/pages/attributesJSON.rst:245]: Link to rigid object doc.

File: docs/pages/attributesJSON.rst [hunk 14]
```diff
@@ -202,18 +320,22 @@ Below are object-specific physical quantities.  These values will override simil
     - integer
 	- The semantic id assigned to objects made with this configuration.
 
+
 `Light Setup Attributes`_
 =========================
-Light Setup Attributes templates hold descriptive information for light setups into Habitat-Sim.  These file names should be formatted as follows:
+Light Setup Attributes templates hold descriptive information for light setups into Habitat-Sim. These file names should be formatted as follows:
```
Comment (dhruvbatra) [inline docs/pages/attributesJSON.rst:326]: Link to "working with lights in Habitat" doc.

File: docs/pages/attributesJSON.rst (hunk 7 above)
Comment (jturner65) [reply docs/pages/attributesJSON.rst:95]: > And what is a semantic scene descriptor file?

The "house" file. Alex and I felt that this was more descriptive of its functionality.

File: docs/pages/attributesJSON.rst [hunk 15]
```diff
@@ -15,7 +15,7 @@ Attributes templates provide a mechanism by which the various constructions in H
 
 `Physics Manager Attributes`_
```
[the remark this answers is not in the document]
Comment (jturner65) [reply docs/pages/attributesJSON.rst:16]: Please [see the thread I started in dev channel to discuss some of these things](https://cvmlp.slack.com/archives/CFN5TAUSD/p1613482950228800)

File: docs/pages/attributesJSON.rst (hunk 2 above)
Comment (aclegg3) [reply docs/pages/attributesJSON.rst:35]: We should upload a terminology page to the documentation.

**Pull request closed without merging**